Sources/ShopifyCheckoutKit/AnalyticsEvents.swift

```diff
@@ -0,0 +1,4594 @@
+    /// page
+    let productVariantViewed: PixelEventsProductVariantViewed?
+    /// The `product_viewed` event logs an instance where a customer visited a
+    /// product details page. This event is available on the product page
+    let productViewed: PixelEventsProductViewed?
+    /// The `search_submitted` event logs an instance where a customer performed a
+    /// search on the storefront. This event is available on the online store page
+    let searchSubmitted: PixelEventsSearchSubmitted?
+
+    enum CodingKeys: String, CodingKey {
+        case cartViewed = "cart_viewed"
+        case checkoutAddressInfoSubmitted = "checkout_address_info_submitted"
+        case checkoutCompleted = "checkout_completed"
+        case checkoutContactInfoSubmitted = "checkout_contact_info_submitted"
+        case checkoutShippingInfoSubmitted = "checkout_shipping_info_submitted"
+        case checkoutStarted = "checkout_started"
+        case collectionViewed = "collection_viewed"
+        case pageViewed = "page_viewed"
+        case paymentInfoSubmitted = "payment_info_submitted"
+        case productAddedToCart = "product_added_to_cart"
+        case productRemovedFromCart = "product_removed_from_cart"
+        case productVariantViewed = "product_variant_viewed"
+        case productViewed = "product_viewed"
+        case searchSubmitted = "search_submitted"
+    }
+}
+
+// MARK: PixelEvents convenience initializers and mutators
+
+extension PixelEvents {
+    init(data: Data) throws {
+        self = try newJSONDecoder().decode(PixelEvents.self, from: data)
+    }
+
+    init(_ json: String, using encoding: String.Encoding = .utf8) throws {
+        guard let data = json.data(using: encoding) else {
+            throw NSError(domain: "JSONDecoding", code: 0, userInfo: nil)
+        }
+        try self.init(data: data)
+    }
+
+    init(fromURL url: URL) throws {
+        try self.init(data: try Data(contentsOf: url))
+    }
+
+    func with(
+        cartViewed: PixelEventsCartViewed?? = nil,
+        checkoutAddressInfoSubmitted: PixelEventsCheckoutAddressInfoSubmitted?? = nil,
+        checkoutCompleted: PixelEventsCheckoutCompleted?? = nil,
+        checkoutContactInfoSubmitted: PixelEventsCheckoutContactInfoSubmitted?? = nil,
+        checkoutShippingInfoSubmitted: PixelEventsCheckoutShippingInfoSubmitted?? = nil,
+        checkoutStarted: PixelEventsCheckoutStarted?? = nil,
+        collectionViewed: PixelEventsCollectionViewed?? = nil,
+        pageViewed: PixelEventsPageViewed?? = nil,
+        paymentInfoSubmitted: PixelEventsPaymentInfoSubmitted?? = nil,
+        productAddedToCart: PixelEventsProductAddedToCart?? = nil,
+        productRemovedFromCart: PixelEventsProductRemovedFromCart?? = nil,
+        productVariantViewed: PixelEventsProductVariantViewed?? = nil,
+        productViewed: PixelEventsProductViewed?? = nil,
+        searchSubmitted: PixelEventsSearchSubmitted?? = nil
+    ) -> PixelEvents {
```

Comment:
I'm not sure we'll need this function? It looks like it returns one of each pixel event